Add info on new libraries (#5220)
segmentio/segment-docs: src/partners/conceptual-model.md

### Mobile Plugins

> Mobile plugins are loaded into [Swift](/docs/connections/sources/catalog/libraries/mobile/apple/destination-plugins/), [Kotlin](/docs/connections/sources/catalog/libraries/mobile/kotlin-android/destination-plugins), [React Native](/docs/connections/sources/catalog/libraries/mobile/react-native/destination-plugins/), [analytics-ios](/docs/connections/sources/catalog/libraries/mobile/ios/#packaging-device-mode-destination-sdks) or [analytics-android](/docs/connections/sources/catalog/libraries/mobile/android/#sending-data-to-destinations).

Reviewer: Putting these in an unordered list helps increase readability!

```suggestion
Mobile plugins are loaded into:
- [Swift](/docs/connections/sources/catalog/libraries/mobile/apple/destination-plugins/)
- [Kotlin](/docs/connections/sources/catalog/libraries/mobile/kotlin-android/destination-plugins)
- [React Native](/docs/connections/sources/catalog/libraries/mobile/react-native/destination-plugins/)
- [analytics-ios](/docs/connections/sources/catalog/libraries/mobile/ios/#packaging-device-mode-destination-sdks)
- [analytics-android](/docs/connections/sources/catalog/libraries/mobile/android/#sending-data-to-destinations).
```

> note ""

> > **Note:** The [Swift](/docs/connections/sources/catalog/libraries/mobile/apple/destination-plugins/), [Kotlin](/docs/connections/sources/catalog/libraries/mobile/kotlin-android/destination-plugins) and [React Native](/docs/connections/sources/catalog/libraries/mobile/react-native/destination-plugins/) libraries were all built with the plugin architecture in mind. This makes adding custom destinations far simpler in comparsion to the older mobile libraries.  

Reviewer: ```suggestion
> **Note:** The [Swift](/docs/connections/sources/catalog/libraries/mobile/apple/destination-plugins/), [Kotlin](/docs/connections/sources/catalog/libraries/mobile/kotlin-android/destination-plugins) and [React Native](/docs/connections/sources/catalog/libraries/mobile/react-native/destination-plugins/) libraries were all built with the plugin architecture in mind. This makes adding custom destinations far simpler than the older mobile libraries.  
```
